newdle/client/src/components/AvailabilityRing.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+AvailabilityRing.defaultProps = {
+  strokeWidth: 10,
+  radius: 50,
+};
+
+export default function AvailabilityRing({
+  available,
+  ifNeeded,
+  totalParticipants,
+  strokeWidth,
+  radius,
+}) {
+  let size = radius * 1.1;
+  let stroke = 7;
+  let normalizedRadius = radius - stroke * 2;
+  let circumference = normalizedRadius * 2 * Math.PI;
+
+  const greenOffset = `rotate(-90 ${size} ${size})`;
+  const yellowOffset = `rotate(${-90 + (available / totalParticipants) * 360} ${size} ${size})`;
+  const redOffset = `rotate(${-90 +
+    ((available + ifNeeded) / totalParticipants) * 360} ${size} ${size})`;
+
+  let style = {
```

Comment:
I'd move this to a CSS module; no need to have it in the JS code

Reply:
I can move this - I originally thought to keep this in the .js because it is less a style and more a functional part of the component, even though it is css.

Reply:
If it was for the circle itself I'd agree, but it's only for the text below so it I'd say it's more styling than a functional part.

Reply:
Oh I misunderstood! I thought you were referring to redOffset, etc.